src/main/kotlin/com/transportes/domain/Truck.kt

```diff
@@ -16,4 +16,14 @@ class Truck(
 ) {
     @Id @GeneratedValue(strategy = GenerationType.UUID)
     lateinit var id: String
+
+    fun copy(
+        brand: String = this.brand,
+        model: String = this.model,
+        patent: String = this.patent
+    ): Truck {
+        val newTruck = Truck(brand, model, patent, transport)
+        newTruck.id = this.id
+        return newTruck
+    }
```

Comment:
```suggestion
    fun update( truckDTO: TruckDTO ) {
        this.brand = truckDTO.brand
        this.model = truckDTO.model
        this.patent = truckDTO.patent
    }
```
Esto funciona pero es me resulta extraño crear un nuevo truck para actualizarlo, podriamos modificar el metodo para recibir el dto y directamente reemplazar los campos de la misma instancia. De esta manera desde el service hacemos:

```kotlin
truck.update( truckDTO )
truckRepository.save( truck )
```